### ISSUE TITLE
Question about LSTM training setting

### ISSUE THREAD

**Issue by harryjhnam** [google-deepmind/mathematics_dataset#15]
I'm trying to reproduce the baseline performance of Attentional LSTM in the paper.

Even though I use the same training hyper-parameters as the paper, I cannot get similar performance as the baseline.
I guess the problem of my implementation is the ignored character.
The paper said 96 characters are used, including one special token. 

What I'm wondering is if there's no <end to sentence> token, should the prediction inferred by the model be also padded with ignored characters? Then the loss function shouldn't ignore <pad> tokens, and I'm worried that the ignored character <pad> gives too much impact for training. 

Also, if it's possible, can you share the training code for baseline results?
It would be greatly helpful to me for reproducing the results of the paper. :)
Thanks!

**Comment by davidsaxton** [comment]
IIRC the answers are fixed-length strings (30 characters?), including padding characters. I.e., the answer to "What is 2 + 2?" is "2··························" where "·" is a special padding character, and the model has to predict the padding characters in addition to the answer. (Note in reality, predicting padding is trivial for the model to do once it's predicted at least one padding character, and does not impact performance.)

Sorry, unable to release training code at the moment.

**Comment by harryjhnam** [comment]
Thank you for your reply! It helped me a lot.
